lib/core/services/search.service.ts

```diff
@@ -32,26 +32,45 @@ export class SearchService {
                 private searchConfigurationService: SearchConfigurationService) {
     }
 
-    async getNodeQueryResults(term: string, options?: SearchOptions): Promise<NodePaging> {
-        const data = await this.apiService.getInstance().core.queriesApi.findNodes(term, options);
+    getNodeQueryResults(term: string, options?: SearchOptions): Observable<NodePaging> {
+        const promise = this.apiService.getInstance().core.queriesApi.findNodes(term, options);
 
-        this.dataLoaded.next(data);
-        return data;
+        promise.then((data: any) => {
+            this.dataLoaded.next(data);
+        })
+
+        return Observable
+            .fromPromise(promise)
+            .catch(err => this.handleError(err));
     }
 
-    async search(searchTerm: string, maxResults: number, skipCount: number): Promise<NodePaging> {
-        const searchQuery = this.searchConfigurationService.generateQueryBody(searchTerm, maxResults, skipCount);
-        const data = await this.apiService.searchApi.search(searchQuery);
+    search(searchTerm: string, maxResults: number, skipCount: number): Observable<NodePaging> {
```

Comment:
Same as above? What is the reason removing "async" here and switching back to Observable?

Reply:
Same as above